cmd/rpcdaemon/commands/trace_api_custom.go

```diff
@@ -0,0 +1,66 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ledgerwatch/turbo-geth/common/hexutil"
+	"github.com/ledgerwatch/turbo-geth/consensus/ethash"
+	"github.com/ledgerwatch/turbo-geth/core/rawdb"
+	"github.com/ledgerwatch/turbo-geth/rpc"
+)
+
+// BlockReward returns the block reward for this block
+func (api *TraceAPIImpl) BlockReward(ctx context.Context, blockNr rpc.BlockNumber) (Issuance, error) {
+	return api.rewardCalc(blockNr, "block") // nolint goconst
+}
+
+// UncleReward returns the uncle reward for this block
+func (api *TraceAPIImpl) UncleReward(ctx context.Context, blockNr rpc.BlockNumber) (Issuance, error) {
+	return api.rewardCalc(blockNr, "uncle") // nolint goconst
+}
+
+// Issuance returns the issuance for this block
+func (api *TraceAPIImpl) Issuance(ctx context.Context, blockNr rpc.BlockNumber) (Issuance, error) {
+	return api.rewardCalc(blockNr, "issuance")
+}
+
+func (api *TraceAPIImpl) rewardCalc(blockNr rpc.BlockNumber, which string) (Issuance, error) {
+	genesisHash := rawdb.ReadBlockByNumber(api.dbReader, 0).Hash()
+	chainConfig := rawdb.ReadChainConfig(api.dbReader, genesisHash)
+	block, err := api.getBlockByRPCNumber(blockNr)
+	if err != nil {
+		return Issuance{}, err
+	}
+	minerReward, uncleRewards := ethash.AccumulateRewards(chainConfig, block.Header(), block.Uncles())
+	issuance := minerReward
+	for _, r := range uncleRewards {
+		p := r // avoids warning?
```

Comment:
It is actually necessary. In the loop above, `r` takes different values on each iteration, but it occupies the same memory. So if you simply take the pointer to it below `&r`, all the pointers will be the same and will be pointing to the last element in the `uncleReward`. Your assignment `p := r` copies the value to `p` and saves your from trouble. Linter saved you from lots of debugging :)

Reply:
Live and learn...